doc/builders/special/mkshellwrapper.section.md

```diff
@@ -0,0 +1,64 @@
+# pkgs.mkShellWrapper {#sec-pkgs-mkShellWrapper}
+
+`pkgs.mkShellWrapper` is a function that generates a wrapper script that
+receives the command to be run as arguments and then run it inside a special
+environment created with [`mkShell`](#sec-pkgs-mkShell).
+
+It's basically like the `--run` flag of `nix-shell`, but using Nix expressions.
+
+## Usage {#sec-pkgs-mkShellWrapper-usage}
+
+```nix
+{ pkgs ? import <nixpkgs> { } }:
+pkgs.mkShellWrapper {
+  shell = pkgs.bash + "/bin/bash"
+
+  name = "demo-environment";
+
+  drv = pkgs.mkShell {
+    buildInputs = with pkgs.python3Packages; [ numpy pandas ];
+  };
+
+  prelude = ''
+    echo Hello world
+  '';
+};
+```
+
+## Attributes {#sec-pkgs-mkShellWrapper-attributes}
+* `shell` (default: `${pkgs.bash} + "/bin/bash"`): Which shell to use as the
+ script shebang.
+* `name` (default: `"${drv.name}-wrapper"`): Name of the derivation for the
+ script.
+* `drv`: Result of [`mkShell`](#sec-pkgs-mkShell). Can be also only the
+ attrset of the arguments for `mkShell` or the list of the packages.
```

Comment:
Would be better to just have users pass drv always.

Reply:
This is the reference behaviour but as most of the usecases are just passing packages in or passing the output of mkShell I think that is useful to support these usecases.

The drv name I got from the function that it's based on (`dockerTools.streamNixShellImage`).